### DISCUSSION TITLE
DEEP MIDI PAGES

### DISCUSSION THREAD

**Discussion by worker-gray** [SynthstromAudible/DelugeFirmware#486] [Ideas]
'DEEP MIDI PAGE'  would implement one row per midi CC and would operate like the timeline. From this page, you could control ALL available midi CC's for it's selected channel, the given channel always being displayed on-screen [i.e. 'CH10' with room for subsequent CC number and value thereof.

An instance of 'Deep Midi Page' could maybe be attached to any given clip OR perhaps be represented by it's own clip (white color audition pad?) With the discrete clip option, it could be copied and sequenced in the arranger like any clip.

The 'Deep Midi Page' would have two views: Expanded and Collapsed

The initial 'deep midi page' would be the expanded view, indicated by every 5th audition pad being lit 1 through 127. Pressing an audition pad would reveal the associated CC on the screen. Each CC would be nested in a different audition pad [scrollable] from the bottom [CC1] to the top [CC127]

The midi channel would be dialed in with the bottom encoder on "pan". All channels (1-16) could be accessed here without exiting to a root menu.

On any given midi channel, an entered note on the CC's 'row' would register as 'on' with the CC value dialed in with the top gold encoder on "level". Holding any entered note would reveal the CC and it's value on the screen. This behavior would be replicated across all channels.

Once all CC data is entered into the DEEP MIDI PAGE, it can be 'collapsed' via a shortcut. The collapsed view will only display the currently used CC's. The same shortcut to 'collapse' would be used to 'expand'.

All entered data would be saved with the song data.

Example of display data for channel: 16, controller: panning, value: left 50%:
CH16
CC10
VL32
The top value would change dependent on the bottom gold encoder's channel selection and would remain on screen.
The bottom two values could either stay on screen until a new one is selected or only display while selected.
This display layout is only a suggestion, as is this entire discussion.

Thank you for your consideration.

**Comment by worker-gray** [comment]
The display could also indicate whether a channel has data in it or not with a box around the channel number or an asterisk. Any indicator would be helpful.

Also, maybe a setup could be saved with an editable name on each channel to indicate what external instrument/device it is controlling.

**Comment by m-m-adams** [comment]
I believe this is already implemented in the automation view unless I'm misunderstanding

**Reply by worker-gray** [reply]
I thought the automation view was internal parameters-- the concept is different anyway-- this would allow you to control ALL 127 midi CC's on a single screen-- each row is a different CC-- and ALL 16 channels

**Reply by seangoodvibes** [reply]
> I thought the automation view was internal parameters-- the concept is different anyway-- this would allow you to control ALL 127 midi CC's on a single screen-- each row is a different CC-- and ALL 16 channels

Automation view lets you control the automation lanes per step for MIDI cc 0-119, cc 120 (pitch wheel), cc 121 (after touch)

Simply go to a MIDI clip, press the clip button and you'll see an overview screen that highlights the MIDI cc's on the pads. Press on the pads to enter the automation lane editing for that MIDI cc.

You can quickly switch from one MIDI cc to the next by going back to the overview (shift + clip / affect entire / audition + clip) or shift + grid shortcut or by turning select.

The overview screen highlights the cc shortcut pads bright white if the MIDI cc's have been automated.

**Reply by worker-gray** [reply]
OK-- i didn't know that. Thanks for the explanation.
